taichi/backends/vulkan/loader.cpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#define VOLK_IMPLEMENTATION
+#include <volk.h>
+
+#include "taichi/backends/vulkan/loader.h"
+
+namespace taichi {
+namespace lang {
+namespace vulkan {
+
+VulkanLoader::VulkanLoader() {
+  initialized = false;
+}
+
+bool VulkanLoader::init() {
+  if (initialized) {
```

Comment:
Use https://en.cppreference.com/w/cpp/thread/call_once ?

Reply:
Hmm, is this mainly because we want to ensure thread safety?

Reply:
exactly :-)